[#13, #23, #30] 가게 API 구현

heesoohi opened the pull request:
- 가게 등록 API
- 가게 리스트 조회 API
- 가게 상세 조회 API

this closes #13 , #23 , #30

heesoohi committed:
feat(store): StoreController, StoreService 기본 구조 생성 (Work in Progress) (#13)

heesoohi committed:
feat(store): store 등록 API 구현 중 (Work in Progress) (#13)

heesoohi committed:
refactor(store): Store 엔티티 DB 제약 사항 추가 (#13)

heesoohi committed:
refactor(store): Store 엔티티에서 필요없는 어노테이션 제거 (#13)

heesoohi committed:
feat(store): 가게 리스트 조회 API 구현 (#23)

heesoohi committed:
feat(store): 가게 상세 조회 API 구현 (#30)

heesoohi committed:
refactor(store): CreateStoreRequest에 유효성 제약사항 추가 및 CreateStoreResponse 생성자 접근제어자 변경 (#13)

heesoohi committed:
refactor(store): GetStoreListResponse 생성자 접근제어자 변경  (#23)

heesoohi committed:
refactor(store): CreateStoreRequest의 유효성 검사 수정 (#13)

iamkixjane reviewed (approved):
수정할 부분이 없습니다!

heesoohi committed:
feat(store): 가게 리스트 조회 시 Pageable 적용 (#23)

subbni reviewed (approved):
좋습니다! 저는 Get 요청의 경우 Response에 따로 prefix를 붙이지 않았는데 이것도 추후 논의후 통일을 해야겠군요 👀

Park-EJ commented on src/main/java/com/example/wait4eat/domain/store/controller/StoreController.java:
```diff
@@ -33,9 +36,11 @@ public ResponseEntity<CreateStoreResponse> create(
+            Pageable pageable
     ) {
-        return ResponseEntity.ok(storeService.getStoreList());
+        return ResponseEntity.ok(storeService.getStoreList(pageable));
```
고생 많으셨습니다! 깔끔한 코드 작성 좋습니다. 의견 하나 드립니당! getStoreList애서 authUser로 권한 검사를 하거나 서비스단에 넘겨주어서 id를 확인하는 게 아니면 @AuthenticationPrincipal이 불필요할 것 같아요~:)

heesoohi replied:
오..! 맞아요! 
SecurityConfig에서 .anyRequest().authenticated()로 설정해놨기 때문에 로그인한 사용자만 접근할 수 있는 상황이긴 합니다..!

@AuthenticationPrincipal AuthUser authUser를 파라미터로 받은 건, SecurityConfig를 확인하지 않아도, 코드만 보고 직관적으로 "이 API는 인증된 사용자만 호출할 수 있다"는 걸 알 수 있게 하려는 의도였어요..! 

컨트롤러 전체의 스타일을 통일하는 것도 좋을 것 같아서, 오늘 저녁 스크럼 때 팀원분들께 한 번 이야기해보면 어떨까요? ㅎㅎ

subbni commented on src/main/java/com/example/wait4eat/domain/store/dto/request/CreateStoreRequest.java:
가게 이미지는 가게 생성 후, 가게 id와 함께 업로드 요청을 받고, S3에 업로드 후 url이 생성될 것 같습니다.
따라서 가게를 생성하는 Request에 imageUrl을 받는 부분은 수정이 필요할 것 같습니다!

subbni commented on src/main/java/com/example/wait4eat/domain/store/dto/request/CreateStoreRequest.java:
또, Store의 depositAmount 필드에 nullable = false 설정이 되어있는 만큼, Request DTO에도 NotNull로 확인해주는 것이 좋을 것 같습니다!

subbni commented on src/main/java/com/example/wait4eat/global/exception/ExceptionType.java:
"사장님은 하나의 가게만 등록할 수 있습니다." 처럼 메세지가 조금 더 명확하면 더더 좋을 것 같습니다 🔥🔥

heesoohi replied:
참고해서 반영했습니다! 😃

subbni reviewed (commented):
깔끔합니다 🙌

heesoohi committed:
refactor(store): Store 생성 및 관련 검증 로직 개선 (#13)
- CreateStoreRequest DTO에서 imageUrl 제거
- DTO의 depositAmount에 NotNull 어노테이션 추가
- 예외 메시지 명확하게 수정

karyosome1 reviewed (approved):
좋습니다!
private int waitingTeamCount; 
이 필드 추가해주실 수 있을까요?
그리고 빌더에도 추가해도 되는지 궁금합니다..

heesoohi committed:
refactor(store): 가게 엔티티에 waitingTeamCount 필드 추가 및 응답 객체 수정 (#13, #23, #30)
- waitingTeamCount 필드를 엔티티 및 가게 조회 응답 객체에 추가
- 가게 응답 객체 필드를 final로 수정

heesoohi committed:
refactor(store): 인증인가가 필요하지 않은 메서드의 불필요한 파라미터 제거 (#23, #30)

heesoohi committed:
refactor(store): 공통 응답 객체 적용 (#13, #23, #30)

kijmane commented on src/main/java/com/example/wait4eat/domain/store/service/StoreService.java:
```diff
@@ -53,11 +54,9 @@ public CreateStoreResponse create(AuthUser authUser, CreateStoreRequest request)
 
     @Transactional(readOnly = true)
-    public List<GetStoreListResponse> getStoreList(Pageable pageable) {
+    public Page<GetStoreListResponse> getStoreList(Pageable pageable) {
```
컨트롤러와 마찬가지로 서비스에서도 응답 포맷 유지하려면PageResponse로 감싸서 반환하는것이 좋을것같습니다!

heesoohi committed:
refactor(store): 응답 객체에서 imageUrl 필드 제거 (#13, #23, #30)

heesoohi committed:
Merge branch 'dev' into feat/#30-store
